slave/cloud-init.sh

```diff
@@ -23,10 +23,17 @@ cat <<"EOF" > $HOME/acceptance.yml
 %(acceptance.yml)s
 EOF
 export FLOCKER_FUNCTIONAL_TEST_CLOUD_CONFIG_FILE=$HOME/acceptance.yml
+export FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER=%(FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER)s
 
-# This will need to change if we ever have latent slaves on non-AWS.
-export FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER=aws
-export FLOCKER_FUNCTIONAL_TEST_AWS_AVAILABILITY_ZONE=$(curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone)
+case "${FLOCKER_FUNCTIONAL_TEST_CLOUD_PROVIDER}" in
```

Comment:
Rather than trying to figure out shell syntax, I wonder if it would be simpler to use jinja2 templates? (Buildbot already depends on them.